pymbolic/geometric_algebra/primitives.py

```diff
@@ -126,7 +126,7 @@ def func(coeff: ArithmeticExpression) -> ArithmeticExpression:
 
     @staticmethod
     @abstractmethod
-    def resolve(expr: Expression) -> Expression:
+    def resolve(expr: ArithmeticExpression) -> ArithmeticExpression:
```

Comment:
I think I had it like that initially too, so fine by me!

The derivative in `pytential` seems to be perfectly happy taking object array and multivectors too?

Reply:
> The derivative in pytential seems to be perfectly happy taking object array and multivectors too?

Yeah. Arguably, that's the point! :facepalm: